### PR TITLE
make security requirements optional

### PR DESCRIPTION
**Pull request by eguzki**
Support for APIProduct obejcts without security (AuthN) configuration

Reconcile adding and removing security from apiproduct

This PR is based on #55, it is ready for review, but should be merged after it

### CODE REVIEW

File: apis/networking/v1beta1/apiproduct_types.go [hunk 1]
```diff
@@ -106,8 +106,11 @@ type APIProductSpec struct {
 	// platform, short-names can also be used instead of a FQDN (i.e. has no
 	// dots in the name). In such a scenario, the FQDN of the host would be
 	// derived based on the underlying platform.
-	Hosts          []string          `json:"hosts"`
-	SecurityScheme []*SecurityScheme `json:"securityScheme"`
+	Hosts []string `json:"hosts"`
+
+	// Configure authentication mechanisms
+	// +optional
```
Comment (guicassolato) [inline apis/networking/v1beta1/apiproduct_types.go:112]: Is this needed for controller-gen to make it optional? I usually rely only on `omitempty` and it does the trick, but I'd love to learn about any other extra advantage.

**Review by guicassolato — APPROVED** [review]
I've left a couple comments kicking off conversation aside, but nothing that disapproves the change of course. LGTM

File: apis/networking/v1beta1/apiproduct_types.go (hunk 1 above)
Comment (eguzki) [reply apis/networking/v1beta1/apiproduct_types.go:112]: I have experienced that too. That's confusing as the [kubebulder doc](https://book.kubebuilder.io/reference/markers/crd-validation.html) says it should be the `'+optional` marker.

To make it explicit and easy to read, I always add it for optional fields

**Comment by eguzki** [comment]
Added fix for the apiproduct status subresource computing logic when resources are optional.

**Pull request merged**